staging/src/k8s.io/component-base/config/testing/helpers.go

```diff
@@ -76,6 +77,9 @@ func matchOutputFile(t *testing.T, actual []byte, expectedFilePath string) {
 		t.Fatalf("couldn't read test data: %v", err)
 	}
 
+	// Windows may have \r\n line endings, which may not match our expectations.
+	expected = []byte(strings.Replace(string(expected), "\r", "", -1))
```

Comment:
what files are getting checked in with \r\n endings? can we fix those fixture files instead of trimming here?

Reply:
Technically, the files have only ``\n`` endings. However, git may checkout files with ``\r\n`` on Windows. One of the typical options for git on Windows is: ``Checkout Windows-style, commit Unix-style line endings``. Files being checked out Windows-style, they have ``\r\n`` file endings.

I think we could enforce the ``\n`` file endings through the ``.gitattributes`` file (https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings).

This would solve the ``TestDefaultsYAML`` test, but will result in the ``TestRoundTripYAML`` to fail instead. That is because ``TestRoundTripYAML`` basically encodes the yaml with ``\r\n``, and then expects the encoding to match the written config file, which will only have ``\n`` in this case. Ultimately, this change addresses both cases. We can update the comment to better explain why it's needed.